addons/stock/models/product.py

```diff
@@ -773,6 +773,13 @@ def write(self, vals):
                 ], order=None, limit=1)
                 if quant:
                     raise UserError(_("This product's company cannot be changed as long as there are quantities of it belonging to another company."))
+                move_lines = self.env['stock.move.line'].search([
+                    ('product_id', 'in', products_changing_company.product_variant_ids.ids),
+                    ('company_id', 'not in', [vals['company_id'], False]),
+                    ('state', '!=', 'cancel'),
+                ], order=None, limit=1)
+                if move_lines:
+                    raise UserError(_("This product's company cannot be changed as long as there are stock move line of it belonging to another company."))
```

Comment:
Shouldn't the check be for `stock.move` instead of `stock.move.line`? Also shouldn't the search be sudo since search is specific to the currently selected company? Also the access error seems to occur even the `state = 'cancel'` (at least in master, didn't test for v14) so may as well drop that condition from the domain?

Otherwise the fix seems fine to me, but can maybe it can be moved before the existing quant check since an existing stock.move is more likely to exist vs a quant. May as well have the failure occur earlier (and possibly avoid an unnecessary second test)

Reply:
The search on `stock.move` is better indeed

Reply:
@ticodoo nice catch! Thanks for the review